packages/core/src/agent/AgentConfig.ts

```diff
@@ -141,6 +141,11 @@ export class AgentConfig {
     return this.initConfig.endpoints as [string, ...string[]]
   }
 
+  public set endpoints(endpoints: string[]) {
+    // if endpoints is not set, return queue endpoint
+    this.initConfig.endpoints = endpoints
+  }
```

Comment:
I think we shouldn't override the init config, as it's an object that the user provided, and we should see it as an immutable object. 

So we should probably add a private property `endpoints` that is initially seeded with the `endpoints` from the init config ,but later can be overwritten. Some generic post on mutation that may be interesting to read: https://blog.sapegin.me/all/avoid-mutation/